docs/usage.rst

```diff
@@ -50,6 +50,12 @@ that in some specific cases `XPath`_ is more powerful than `CSS`_.
 .. _XPath: http://www.w3.org/TR/xpath
 .. _CSS: http://www.w3.org/TR/selectors
 
+.. note::
+    HTML comments mentioned within <style> and <script> tags will be treated as raw text.
+    The reason is that it is a common practice to comment out the content of <style> 
```

Comment:
Hm, this isn't quite right -- we don't do any special handling because of this.

The reason why comments inside these tags aren't parsed as XML elements is that the content of `<style>` and `<script>` are meant to be application-specific (that is, it is not a concern of the HTML/XML parser, because there can be _anything_ inside of it, users can use their own styling or scripting language).

Reply:
Hello @harshasrinivas, thanks for putting up with all my nitpicking so far!
Are you interested in following up on this? :)